app/models/category.rb

```diff
@@ -343,7 +343,7 @@ def ensure_slug
       # if we don't unescape it first we strip the % from the encoded version
       slug = SiteSetting.slug_generation_method == 'encoded' ? CGI.unescape(self.slug) : self.slug
       # sanitize the custom slug
-      self.slug = Slug.sanitize(slug)
```

Comment:
I think that was the last use of `Slug.sanitize`. I don't see any plugins using it either. Can we remove or deprecate it? 🙂 (in this or a separate PR, either way)

Reply:
I do not see it used anywhere. What about removing it completely? :)

Reply:
Yes, that exactly what I did on https://github.com/discourse/discourse/pull/11225/files but the case sensitive-ness of categories slug was breaking it, so that is the whole reason this discussion started!

Reply:
Unfortunately, we cannot get rid of `Slug.sanitize`, but I renamed it to `Slug.for_category`. We need it because `slug_generation_method` should be ignored for categories.